### PR TITLE
Inference server for timl

### PR DESCRIPTION
**Pull request by ivanzvonkov**
This PR adds timl inference deployment! 🚀

Essentially this means that files in `crop_classification/models/*.pt` will be deployed to Google Cloud on each main build.

Components:
- A torchserve handler for running deployed timl predictions
- A Dockerfile for housing the torchserve handler
- A Google Cloud function for triggering inference when a new file is uploaded to a specified bucket
- A script for deploying the docker container and the Google Cloud function to Google Cloud
- An addition to ci.yml to run the above deployment script during CI

Completes remaining tasks in #4

### CODE REVIEW

**Comment by gabrieltseng** [comment]
Nice ! This is awesome; thanks @ivanzvonkov .

It seems like the finetuning will need to happen locally? If so, we should get some script which finetunes for a task (crop type & area) and saves a jit model

**Comment by ivanzvonkov** [comment]
@gabrieltseng  Thanks!

Yes exactly! I think it would make a lot of sense for this script to be similar to https://github.com/nasaharvest/crop-mask/blob/master/scripts/model_train.py 

Is this something you could tackle by any chance?

**Comment by gabrieltseng** [comment]
> Is this something you could tackle by any chance?

Yes - what model naming convention should be followed? `<region>_<target_class>`?

**Comment by ivanzvonkov** [comment]
> > Is this something you could tackle by any chance?
> 
> Yes - what model naming convention should be followed? `<region>_<target_class>`?

Thank you! I'd probably add year: `<region>_<target_class>_<year>` where year represents the start year of the model evaluation data.

File: crop_classification/torchserve_handler.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    def initialize(self, context):
+        print("HANDLER: Initializing handler")
+        super().initialize(context)
+        properties = context.system_properties
+        model_dir = properties.get("model_dir")
+        sys.path.append(model_dir)
+
+        # TODO need to pass the normalizing dict here somehow
+        print("HANDLER: Initializing inference module")
+        self.inference_module = Inference(model=self.model, normalizing_dict=None)
```
Comment (ivanzvonkov) [inline crop_classification/torchserve_handler.py:61]: @gabrieltseng The easiest way that I have found to pass the normalizing_dict here is by making it an attribute of the model. (Note jit can only have primitive attributes so normalizing dict must be `Dict[str, List[float]]` instead of `Dict[str, List[np.ndarray]]`

I think this would make sense to set when running the fine-tuning script described above. Thoughts? We can discuss on Friday too

**Comment by ivanzvonkov** [comment]
Verified automatic deployment here: https://github.com/nasaharvest/timl/actions/runs/1984275429

**Review by gabrieltseng — APPROVED** [review]
